tests/models/test_gemini.py

```diff
@@ -1021,6 +1022,83 @@ async def get_image() -> BinaryContent:
     )
 
 
+@pytest.mark.parametrize(
+    'url',
+    [
+        pytest.param(AudioUrl(url='https://cdn.openai.com/API/docs/audio/alloy.wav'), id='AudioUrl'),
+        pytest.param(
+            DocumentUrl(url='https://storage.googleapis.com/cloud-samples-data/generative-ai/pdf/2403.05530.pdf'),
+            id='DocumentUrl',
+        ),
+        pytest.param(
+            ImageUrl(url='https://upload.wikimedia.org/wikipedia/commons/6/6a/Www.wikipedia_screenshot_%282021%29.png'),
+            id='ImageUrl',
+        ),
+        pytest.param(VideoUrl(url='https://data.grepit.app/assets/tiny_video.mp4'), id='VideoUrl'),
+    ],
+)
+async def test_url_inputs_are_downloaded(
+    url: AudioUrl | DocumentUrl | ImageUrl | VideoUrl,
+    client_with_handler: ClientWithHandler,
+    env: TestEnv,
+    allow_model_requests: None,
+) -> None:
+    def handler(request: httpx.Request) -> httpx.Response:
```

Comment:
Instead of directly inspecting these requests and faking the responses, can we use `pytest.mark.vcr` to actually perform these requests once and record them, and then verify that the model output is as expected?

Reply:
@DouweM do you have any tips on how to deal with VertexAI and cassettes? the cassettes I uploaded won't work because the CI is not properly configured for Vertex AI as far as I understand

The google-GLA cassette works instead.